scripts/file_resume.js

```diff
@@ -33,41 +33,41 @@ $(function() {
     });
 
     function showProgress(text) {
-        $("#upload_progress").html(text);
+        document.getElementById("upload_progress").innerHTML = text;
     }
 
     if(resumable.support) {
-        resumable.assignBrowse($("#resumable_browse"));
-        resumable.assignDrop($("#upload_form"));
+        resumable.assignBrowse(document.getElementById("resumable_browse"));
+        resumable.assignDrop(document.getElementById("upload_form"));
 
         // Show the resumable div
-        $("#resumable_uploader").show();
+        this.document.getElementById("resumable_uploader").style.display = "";
         // Hide the old uploader
-        $("#old_uploader").hide();
+        document.getElementById("old_uploader").style.display = "none";
     }
 
     // Before we start the upload, prevent the user from hitting upload again.
-    $("#old_submit").click(function(ev) {
-        let file = $("#old_browse")[0].files[0];
+    document.getElementById("old_submit").addEventListener("click", function(ev) {
+        let file = document.getElementById("old_browse").files[0];
         if(file.size >= maxNormSize) {
             alert(uploadMessages.fileTooBig);
             ev.preventDefault();
             return false;
         }
         if(validate(file.name)) {
             // won't work if we disable browse button so hide it
-            $("#old_browse").hide();
-            $("#old_submit").prop( "disabled", true);
+            document.getElementById("old_browse").style.display = "none";
```

Comment:
No need to explain the reason (I'm a js/jQuery noob) but sometimes it seems like `.hide()/show()` is replaced with `.style.display = "none"`/`style.display = ''`, sometimes with `style.display = ''`, sometimes adding/removing "nodisp", and sometimes a bit of one method and a bit of the other. If that's all OK, I'm happy.

Reply:
good question. I used the approach that matched what the php output to hide the elements. Some php pages output elements with the "nodisp" class, others set in the stylesheet `display: none`. I just kept the js consistent with the existing page.

Reply:
This seems slightly risky. Jquery hide() doc. says "the value of the display property is saved in jQuery's data cache so that display can later be restored to its initial value". If the php were ever changed to have a different display value then the js function would no longer work correctly if the display value is built into it.

Reply:
Yeah, the jquery hide/show has to be generic. This code only has to handle the specific elements it is working with. I confirmed the mechanism they were using to hide and implemented it here. The initial value is '' for display so we can use that to show, and 'none' to hide. Someone can correct me if I'm wrong based on testing.